transports/transports.go

```diff
@@ -77,7 +77,9 @@ func ListNames() []string {
 	defer kt.mu.Unlock()
 	var names []string
 	for _, transport := range kt.transports {
-		names = append(names, transport.Name())
+		if !transport.Deprecated() {
```

Comment:
Bottom line is to get "atomic" out of the list of all transports, while still continuing to support it.
If you want to add a constant that lists deprecated transports and then to see if the transport.Name() is in the list, I am fine with that.